src/core/logger.py

```diff
@@ -0,0 +1,42 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+from datetime import datetime
+
+LOG_FILE_NAME = "Sammo.log"
+
+
+class Logger:
```

Comment:
+1 to add a `Logger` class :+1:

But the good way for logging in QGIS is to use a dedicated tab in the log panel. For example with: `QgsMessageLog.logMessage(msg, "Sammo-Boat")`. So the `_log` method would look like:

``` python
def _log(prefix: str, msg: str):
    msg = "{}: {}".format(prefix, msg)
   QgsMessageLog.logMessage(msg, "Sammo-Boat")
```

What do you think?